packages/apps/stock/src/components/templates/internalMove/InternalMoveCreationAlert/InternalMoveCreationAlert.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Axelor Business Solutions
+ *
+ * Copyright (C) 2023 Axelor (<http://axelor.com>).
+ *
+ * This program is free software: you can redistribute it and/or  modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useState} from 'react';
+import {StyleSheet, View} from 'react-native';
+import {useTranslator} from '@axelor/aos-mobile-core';
+import {
+  Alert,
+  CheckboxScrollList,
+  Text,
+  useThemeColor,
+} from '@axelor/aos-mobile-ui';
+
+interface InternalMoveCreationAlertProps {
+  isAlertVisible: boolean;
+  setIsAlertVisible: (visible: boolean) => void;
+  lines: any[];
+  setLines: (lines: any[]) => void;
+}
+
+const InternalMoveCreationAlert = ({
+  isAlertVisible,
+  setIsAlertVisible,
+  lines,
+  setLines,
+}: InternalMoveCreationAlertProps) => {
+  const Colors = useThemeColor();
+  const I18n = useTranslator();
+
+  const [selectedLines, setSelectedLines] = useState([]);
+
+  const renderChexboxItem = ({item}) => {
+    return (
+      <View style={styles.renderContainer}>
+        <View style={styles.renderFirstLine}>
+          <Text writingType="important">{item.product?.name}</Text>
+          <Text>
+            {item.realQty} {item.unit?.name}
+          </Text>
+        </View>
+        {item.trackingNumber?.trackingNumberSeq && (
+          <Text writingType="details">
+            {item.trackingNumber?.trackingNumberSeq}
+          </Text>
+        )}
+      </View>
+    );
+  };
+
+  return (
+    <Alert
+      visible={isAlertVisible}
+      title={I18n.t('Stock_Products')}
+      cancelButtonConfig={{
+        showInHeader: true,
+        onPress: () => setIsAlertVisible(false),
+      }}
+      confirmButtonConfig={{
+        title: I18n.t('Stock_Remove'),
+        iconName: 'trash3-fill',
+        width: '45%',
+        color: Colors.warningColor,
+        onPress: () => {
+          setLines(
+            lines.filter(
+              _line =>
+                selectedLines.filter(
+                  _selectedLine => _selectedLine.id === _line.id,
+                ).length === 0,
+            ),
+          );
```

Comment:
If product list is reset when in toStockLocation step (all lines were removed), we should go back to addLines step as we should not create an internal move without lines. Currently, we can still select a toStockLocation and click on Realize button